tests/api_test.py

```diff
@@ -2076,6 +2076,17 @@ def f(_, __):
       # level, which is no longer live.
       jax.jit(jnp.add)(jnp.ones(()), count)
 
+  def test_escaped_tracer_transform_name(self):
+    with self.assertRaisesRegex(core.UnexpectedTracerError,
+                                "transformed by jit"):
+      jax.jit(self.helper_save_tracer)(1)
+      _ = self._saved_tracer+1
```

Comment:
Here and throughout, I think the line after the error will not be executed, because the block stops execution when it sees the (expected) error. To make sure the cleanup step runs regardless of whether the expected error is raised, we might consider a `try...finally` statement

Reply:
Wait, strike that: I see now this final line is actually where the error is occurring! Carry on... 😬